Common++/header/IpAddress.h

```diff
@@ -110,6 +110,14 @@ namespace pcpp
 		 */
 		bool operator!=(const IPv4Address& rhs) const	{ return !(*this == rhs); }
 
+		/**
+		 * Checks whether the address matches a subnet.
+		 * For example: if subnet is 10.1.1.1/24 and address is 10.1.1.9 then the method will return true
+		 * Another example: if subnet is 10.1.1.1/16 and address is 11.1.1.9 then the method will return false
+		 * @param[in] subnet A string in X.X.X.X/Y format representing the masked subnet to compare with the address
+		 */
+		bool matchSubnet(const std::string& subnet) const;
```

Comment:
whould be nice to introduce a proper IPv4Network class to check the CIDR notation is valid.

But it's a bit out of scope for this MR

Reply:
So can we merge?

Reply:
@yishai1999 I'm not sure if you saw the remaining comments. Let's fix them and merge the PR?

Reply:
Yup, I'm on it.

Reply:
I fixed whatever you commented about but the mac and openBSD CI jobs don't pass. Any clue why?